lambdas/shared/t4_lambda_shared/preview.py

```diff
@@ -59,6 +60,27 @@ def decompress_stream(chunk_iterator, compression):
             break
 
 
+def extract_excel(file_, as_html=False):
+    """
+    excel file => data frame => html
+    Args:
+        file_ - file-like object opened in binary mode, pointing to XLS or XLSX
+    Returns:
+        body - html or text version of *first sheet only* in workbook
+        info - metadata
+    """
+    try:
+        first_sheet = pandas.read_excel(file_, sheet_name=0)
+    except XLRDError:
+        first_sheet = pandas.read_excel(file_, sheet_name=0, engine='openpyxl')
+
+    if as_html:
+        html = remove_pandas_footer(first_sheet._repr_html_())  # pylint: disable=protected-access
```

Comment:
Consider using [`DataFrame.to_html()`](https://pandas.pydata.org/pandas-docs/version/1.1.5/reference/api/pandas.DataFrame.to_html.html). It looks like `df._repr_html_() == df.to_html(notebook=True)`.

@nl0 
I noticed that `df._repr_html_()` outputs inline CSS, don't we do styling on frontend? If we do, we could omit CSS with `df.to_html(notebook=False)`.

Reply:
i need to check and see which styles are provided by the backend and which -- by the front-end. ideally we should have all the styles provided by the frontend, yep, but let's backlog this. it would be very helpful to see isolated output of `df.to_html(notebook=True)` and `df.to_html(notebook=False)` for some sample file.

Reply:
i switched to the public method in this diff but there are still other places in the code that use the private one

Reply:
(actually switching to `to_html(notebook=True)` breaks a unit test in preview so I'm not gonna mess with it right now)